site/docs/types/compound_logical_types.md

```diff
@@ -1,15 +1,15 @@
-# Compound Types
+# Compound Logical Types
 
 Compound types include any type that is configurable including complex types as well as configurable scalar types.
 
-| Type Name                   | Description                                                  | Arrow Analog        | Iceberg Analog | Spark Analog   | Trino Analog                |
-| --------------------------- | ------------------------------------------------------------ | ------------------- | -------------- | -------------- | --------------------------- |
-| FIXEDCHAR&lt;L&gt;    | A fixed-length field of length L. L can be between [1..2,147,483,647]. Values shorter than the length of the field are padded with spaces. | -                   | -              | CharType(L)    | CHAR(L)                     |
-| VARCHAR&lt;L&gt;         | A field that can holds UTF-8-encoded strings between 0 and L in character length. The value of L can be between [1..2,147,483,647]. Values shorter than L are not padded. | -                   | -              | VarcharType(L) | VARCHAR(L)                  |
-| FIXEDBINARY&lt;L&gt;     | A binary field that is fixed in width to L bytes. Values that are shorter than L are 0-padded. | FixedSizeBinary&lt;L&gt;  | FIXED&lt;L&gt;       | -              | -                           |
-| DECIMAL&lt;P, S&gt;   | A fixed-precision decimal value having precision (P, number of digits) <= 38 and scale (S, number of fractional digits) 0 <= S <= P. | Decimal&lt;P, S, bitwidth=128&gt; | DECIMAL(P,S)   | DECIMAL(P,S)   | DECIMAL(P,S)                |
-| STRUCT&lt;T1,...,T2&gt; | A list of types in a defined order. | struct_&lt;*&gt;                  | struct&lt;*&gt; | struct&lt;*&gt; | row&lt;*&gt;   |
-| NSTRUCT&lt;N:T1,...,N:T2&gt; | **Pseudo-type**: A struct that maps unique names to value types. Each name is a UTF-8-encoded string. Each value can have a distinct type. Note that NSTRUCT is actually a pseudo-type. This is because Substrait's core type system is based entirely on ordinal positions, not named fields. Nonetheless, when working with systems outside Substrait, names are important. | struct_&lt;*&gt; | struct&lt;*&gt; | struct&lt;*&gt; | row&lt;*&gt; |
-| LIST&lt;T&gt;               | A list of values of type T. The list can be between [0..2,147,483,647] values in length. | list                | list           | list           | array                       |
-| MAP&lt;K, V&gt;                   | An unordered list of type K keys with type V values.         | map&lt;k,v&gt;            | map&lt;k,v&gt;       | -              | map&lt;k,v&gt;                    |
+| Type Name                    | Description                                                  | Canonical physical representation (protobuf)
```

Comment:
I named it that way because:

 - type variation specified: storage format should be whatever corresponds to the variation;
 - type variation not specified: storage format is the "canonical" format.

My reasoning is that once we elaborate on a communication protocol, there should be some canonical data format, even if it's only useful for testing because it isn't very efficient. By precedence, the most logical choice for this format seems to me to be the virtual table format. Writing that format down was my intention for the column.

Where exactly am I going wrong with that line of reasoning?

Reply:
I think the issue here is canonical. Substrait has not opinions about the default or best representation of data. Using the word canonical sounds too much like "recommended" or "preferred" or ideal. Substrait doesn't portend to have any opinion about such things. So in most cases, 99.999% of all data related to a substrait plan has no default or standard representation. The only place where Substrait has to have an opinion is when it stores literals. This is unfortunate but unavoidable. But this is probably 0.001% or less of all data relating to a Substrait plan. Saying that the 0.001% part of things is "canonical" is misleading.

For reference, even in protobuf, the physical representation of int32 isn't actually what most people would think of as a int32 values. Instead, it's a Google varint encoded integer. The only thing int32 is the name of the type in the context of the protobuf idl and how the varint encoded value is exposed at the api level. So calling that a physical representation is also weird.